Signal/src/ViewControllers/HomeViewController.m

```diff
@@ -287,6 +287,22 @@ - (void)viewDidLoad
     [self updateBarButtonItems];
 }
 
+- (void)viewDidAppear:(BOOL)animated
+{
+    [super viewDidAppear:animated];
+
+    // Keep in mind viewDidAppear is called while the app is in the background if the app was
+    // launched by a voip notification. This is fine - it will remain visible to the user
+    // when they eventually launch the app, but we shouldn't make any changes assuming
```

Comment:
// it will remain visible to the user when they eventually launch the app

Is that true?  What if:

* App is launched from voip.
* We show upgrade splash & mark all upgrades as "seen".
* App - still in the background - is killed by iOS.
* User launches app, never sees the upgrade.

Reply:
Should we observe "app lifecycle events" and only show & mark as seen the upgrade UI when they are actually visible?

Reply:
I took a different approach, which seemed simpler, PTAL @charlesmchen